source/unix/openssl_aes.c

```diff
@@ -0,0 +1,741 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/cal/symmetric_cipher.h>
+
+#include <openssl/evp.h>
+
+struct openssl_aes_cipher {
+    struct aws_symmetric_cipher cipher_base;
+    EVP_CIPHER_CTX *encryptor_ctx;
+    EVP_CIPHER_CTX *decryptor_ctx;
+    struct aws_byte_buf working_buffer;
+};
+
+static int s_encrypt(
+    struct aws_symmetric_cipher *cipher,
+    const struct aws_byte_cursor *input,
+    struct aws_byte_buf *out) {
+    size_t required_buffer_space = input->len + cipher->block_size;
+    size_t available_write_space = out->capacity - out->len;
+
+    if (available_write_space < required_buffer_space) {
+        if (aws_byte_buf_reserve_relative(out, required_buffer_space) != AWS_OP_SUCCESS) {
+            return aws_raise_error(AWS_ERROR_SHORT_BUFFER);
+        }
+    }
+
+    available_write_space = out->capacity - out->len;
+    struct openssl_aes_cipher *openssl_cipher = cipher->impl;
+
+    int len_written = (int)(available_write_space);
+    if (!EVP_EncryptUpdate(
+            openssl_cipher->encryptor_ctx, out->buffer + out->len, &len_written, input->ptr, (int)input->len)) {
```

Comment:
naively casting `input->len` to `int`
- raise error if `len` is too big
- OR run this in a loop, doing the work in chunks

the [docs](https://www.openssl.org/docs/man3.1/man3/EVP_EncryptUpdate.html) say:
> the amount of data written can be anything from zero bytes to (inl + cipher_block_size) bytes

so... I think the max amount of work we can do per call is `INT_MAX - cipher_block_size`???  (or pick something smaller that doesn't push up so hard against the limits)

Reply:
I made it check in the base class before any of these functions are called to check the input. That way the behavior is the same across implementations. Added a unit test as well.